### PR TITLE
CI: Build Thunks

### PR DESCRIPTION
**Pull request by Sonicadvance1**
To ensure that our thunks don't become horribly broken with our changes. Build them in CI.

Requires some tinkering because the 20.04 requirement, but otherwise works.

### CODE REVIEW

File: ThunkLibs/GuestLibs/CMakeLists.txt [hunk 1]
```diff
@@ -41,8 +42,10 @@ function(generate NAME SOURCE_FILE)
       DEPENDS "${GENERATOR_EXE}"
       DEPENDS "${SOURCE_FILE}"
       COMMAND "${GENERATOR_EXE}" "${SOURCE_FILE}" "${NAME}" "-${WHAT}" "${OUTFILE}" -- -std=c++17 -DGUEST_THUNK_LIBRARY
-            # Expand include directories to space-separated list of -isystem parameters
-           "$<$<BOOL:${prop}>:;-isystem$<JOIN:${prop},;-isystem>>"
+        # Expand compile definitions to space-separated list of -D parameters
```
Comment (skmp) [inline ThunkLibs/GuestLibs/CMakeLists.txt:45]: Why is this needed?

File: ThunkLibs/libX11/libX11_Guest.cpp [hunk 2]
```diff
@@ -206,7 +206,10 @@ extern "C" {
 
     MakeHostFunctionGuestCallable(ret->resource_alloc);
     MakeHostFunctionGuestCallable(ret->idlist_alloc);
+#if ((X11_VERSION_MAJOR << 24) | (X11_VERSION_MINOR << 16) | (X11_VERSION_PATCH)) >= ((1 << 24) | (7 << 16) | (0))
```
Comment (skmp) [inline ThunkLibs/libX11/libX11_Guest.cpp:209]: (X11_VERSION_MAJOR >= 1 && X11_VERSION_MINOR >= 16  && X11_VERSION_PATCH >= 0) ?

File: ThunkLibs/GuestLibs/CMakeLists.txt (hunk 1 above)
Comment (Sonicadvance1) [reply ThunkLibs/GuestLibs/CMakeLists.txt:45]: compile definitions automatically get stripped of their `-D` in cmake, so we need to add them back in to use this interface.

File: ThunkLibs/libX11/libX11_Guest.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply ThunkLibs/libX11/libX11_Guest.cpp]: Resolved.

**Review by skmp — APPROVED** [review]

File: ThunkLibs/GuestLibs/CMakeLists.txt [hunk 3]
```diff
@@ -28,6 +28,7 @@ function(generate NAME SOURCE_FILE)
   # Shorthand for the include directories added after calling this function.
```
Comment (neobrain) [inline ThunkLibs/GuestLibs/CMakeLists.txt:28]: Since we now can propagate definitions, we can remove the manual `-DGUEST_THUNK_LIBRARY` switch below and in `add_guest_lib` and instead use this:

```suggestion
  target_compile_definitions(${NAME}-guest-deps INTERFACE GUEST_THUNK_LIBRARY)
  # Shorthand for the include directories added after calling this function.
```

**Review by neobrain — APPROVED** [review]
Glad 20.04 doesn't require any major changes here. Thanks for wiring things up with CI!

File: ThunkLibs/GuestLibs/CMakeLists.txt (hunk 3 above)
Comment (Sonicadvance1) [reply ThunkLibs/GuestLibs/CMakeLists.txt:29]: Good shout, resolved.
